src/storage/page_buffer.c

```diff
@@ -7856,26 +7859,24 @@ pgbuf_request_data_page_from_page_server (const VPID * vpid)
   /* Send a request to Page Server for the Page. */
   if (get_server_type () == SERVER_TYPE_TRANSACTION)
     {
-      constexpr size_t PAGEID_SIZE = sizeof (VPID::pageid);
-      constexpr size_t VOLID_SIZE = sizeof (VPID::volid);
-      char buffer[PAGEID_SIZE + VOLID_SIZE + sizeof (LOG_LSA)];
+      cubpacking::packer pac;
+      int size = 0;
 
-      std::memcpy (buffer, &(vpid->pageid), sizeof (vpid->pageid));
-      size_t bytes_copied = sizeof (vpid->pageid);
-
-      std::memcpy (buffer + bytes_copied, &(vpid->volid), sizeof (vpid->volid));
-      bytes_copied += sizeof (vpid->volid);
+      size += cublog::lsa_utils::get_packed_size(pac, size);
+      size += vpid_utils::get_packed_size(pac, size);
+      std::unique_ptr<char[]> buffer (new char[size]);
```

Comment:
If both cublog::lsa_utils::get_packed_size and vpid_utils::get_packed_size would get constexpr specifier, this could be rewritten as:

```suggestion
      constexpr size_t VPID_SIZE = vpid_utils::get_packed_size (pac, 0);
      constexpr size_t MSG_SIZE = VPID_SIZE + cublog::lsa_utils::get_packed_size (pac, size);
      char buffer[MSG_SIZE];
```